docusaurus/docs/engineering/knowledge-base/development-lifecycle.md

```diff
@@ -4,154 +4,239 @@ sidebar_position: 7
 
-3. Run the following commands from your terminal, in the root folder of the repo that you want to upload.
+1. **Keep your code up to date.**
 
-```bash   showLineNumbers
-git config --get remote.origin.url | xargs git remote add upstream
-git remote set-url origin YOUR_NEW_REMOTE_URL_HERE
-git fetch
-```
+   > You did the thing and now you (and everyone else with a copy of the project on their PC) should get the new and updated version of the code - the one that includes your latest edits! In technical terms, you need to 'pull' the changes.
+
+## Guide
+
+The following is a more in-depth guide to the steps one would follow to contribute a simple change to one of Boba's codebases.
+
+:::note
+If you've looked at our [Recommended Tools](/docs/engineering/knowledge-base/recommended-tools.md) document, you've seen we recommend Visual Studio Code as a code editor. One reason for this is its excellent integration with git. This lets you perform various tasks using the editor's user interface instead of typing commands in your terminal. Feel free to use integrations if you prefer. However, we will still list the commands here so that everyone can follow along and understand what's happening behind the scenes.
+:::
+
+### 1. Fork the repository on GitHub
+
+Forking creates an exact copy of the entire codebase to your own GitHub account. This means all it's files and the history of changes. You own this copy (in terms of permissions) and it's completely separate from the original. No matter what changes you make, the original codebase (often referred to as `upstream`) will remain unchanged.
+
+You can fork a repository on GitHub from its web interface by going to the repository's URL. For example, the repository for this documentation is `https://www.github.com/BobaBoard/boba-docs`, where `BobaBoard` is the name of the owner of the repository, and `boba-docs` is the repository's name.
+
+Once you've created a fork, this copy will live somewhere that looks like so: `https://www.github.com/<YOUR_GITHUB_USERNAME>/<REPOSITORY_NAME>`
 
-:::warning
-If you want to use a different GitHub account for BobaBoard, your `origin` URL
-will be different. Follow the instructions in the ["use a different GitHub account" section](#extra-use-a-different-github-account).
+:::info
+The name of your fork doesn't have to match the name of the original project. For this guide, however, we'll assume you have kept the same name.
 :::
 
-At this point, you can proceed with the regular flow for a pull request.
+### 2. Clone your fork to your machine
 
-### 1. Create a branch off of `master` for your new feature
+This means copying the code that is stored in your GitHub account to your PC. It doesn't really matter where you want to do this (make sure you have permissions?), though you might want to create a folder to keep all your BobaBoard-related coding in. For example: `/home/NAME/code-projects/bobaboard`.
```

Comment:
"It doesn't really matter where you want to do this, as long as you have permission to create files. For this project, we suggest suggest creating a folder..."